### PR TITLE
Fix flakey test timeouts

### PR DESCRIPTION
**Pull request by JamesNK**
Fix timeouts in tests. Add additional logging.

### CODE REVIEW

**Comment by JamesNK** [comment]
Error from https://github.com/grpc/grpc-dotnet/pull/1387

**Comment by jtattermusch** [comment]
@captainsafia  can you please review?

File: test/FunctionalTests/Client/HedgingTests.cs [hunk 1]
```diff
@@ -289,7 +289,7 @@ Task DuplexDeadlineExceeded(IAsyncStreamReader<DataMessage> requestStream, IServ
             var client = TestClientFactory.Create(channel, method);
 
             // Act
-            var call = client.DuplexStreamingCall(new CallOptions(deadline: DateTime.UtcNow.AddMilliseconds(300)));
+            var call = client.DuplexStreamingCall(new CallOptions(deadline: DateTime.UtcNow.AddMilliseconds(500)));
```
Comment (captainsafia) [inline test/FunctionalTests/Client/HedgingTests.cs:292]:
```suggestion
            var deadlineTimeout = 500;
            var call = client.DuplexStreamingCall(new CallOptions(deadline: DateTime.UtcNow.AddMilliseconds(deadlineTimeout)));
```

File: test/FunctionalTests/Web/Server/DeadlineTests.cs [hunk 2]
```diff
@@ -86,6 +86,8 @@ async Task<HelloReply> WaitUntilDeadline(HelloRequest request, ServerCallContext
                 return false;
             });
 
+            using var httpEventSource = new HttpEventSourceListener(LoggerFactory);
```
Comment (captainsafia) [inline test/FunctionalTests/Web/Server/DeadlineTests.cs:89]: Don't think this is needed here.

```suggestion
```

File: test/FunctionalTests/Web/Server/DeadlineTests.cs [hunk 3]
```diff
@@ -86,6 +86,7 @@ async Task<HelloReply> WaitUntilDeadline(HelloRequest request, ServerCallContext
                 return false;
             });
 
+
```
Comment (captainsafia) [inline test/FunctionalTests/Web/Server/DeadlineTests.cs:89]:
```suggestion
```

**Review by captainsafia — APPROVED** [review]
Approved with some minor fixes.

**Pull request merged**